Code/Framework/AzQtComponents/AzQtComponents/Components/FancyDockingDropZoneWidget.h

```diff
@@ -31,58 +31,57 @@ namespace AzQtComponents
 
         // Constant for the dock width/height size (in pixels) before we need to start
         // scaling down the drop zone sizes, or else they will overlap with the center
         // tab icon or each other
-        int minDockSizeBeforeDropZoneScalingInPixels;
+        static constexpr int minDockSizeBeforeDropZoneScalingInPixels = dropZoneSizeInPixels * 3;
 
         // Constant for the factor by which we must scale down the drop zone sizes once
         // the dock width/height size is too small
-        qreal dropZoneScaleFactor;
+        static constexpr qreal dropZoneScaleFactor  = 0.25;
 
         // Constant for the percentage to scale down the inner drop zone rectangle for the center tab drop zone
-        qreal centerTabDropZoneScale;
+        static constexpr qreal centerTabDropZoneScale  = 0.5;
 
         // Constant for the percentage to scale down the center tab drop zone for the center tab icon
-        qreal centerTabIconScale;
+        static constexpr qreal centerTabIconScale  = 0.5;
 
         // Constant for the drop zone hotspot default color
-        QColor dropZoneColor;
+        static const QColor dropZoneColor;
 
         // Constant for the drop zone border color
-        QColor dropZoneBorderColor;
+        static const QColor dropZoneBorderColor;
 
         // Constant for the border width in pixels separating the drop zones
-        int dropZoneBorderInPixels;
+        static constexpr int dropZoneBorderInPixels  = 1;
 
         // Constant for the border width in pixels separating the drop zones
-        int absoluteDropZoneSizeInPixels;
+        static constexpr int absoluteDropZoneSizeInPixels  = 25;
 
         // Constant for the delay (in milliseconds) before a drop zone becomes active
         // once it is hovered over
-        int dockingTargetDelayMS;
+        static constexpr int dockingTargetDelayMS = 110;
 
         // Constant for the rate at which we will update (fade in) the drop zone opacity
         // when hovered over (in milliseconds)
-        int dropZoneHoverFadeUpdateIntervalMS;
+        static constexpr int dropZoneHoverFadeUpdateIntervalMS  = 20;
 
         // Constant for the incremental opacity increase for the hovered drop zone
         // that will fade in to the full drop zone opacity in the desired time
-        qreal dropZoneHoverFadeIncrement;
+        static constexpr qreal dropZoneHoverFadeIncrement = dropZoneOpacity / (dockingTargetDelayMS / dropZoneHoverFadeUpdateIntervalMS);
 
         // Constant for the path to the center drop zone tabs icon
-        QString centerDropZoneIconPath;
-
-        FancyDockingDropZoneConstants();
+        static const QString centerDropZoneIconPath;
 
+        FancyDockingDropZoneConstants() = delete;
         FancyDockingDropZoneConstants(const FancyDockingDropZoneConstants&) = delete;
         FancyDockingDropZoneConstants& operator=(const FancyDockingDropZoneConstants&) = delete;
     };
```

Comment:
Hmm I'm not sure about making all these `static` and `constexpr`, it prevents us from ever having another instance of this type... I'll defer to @cgalvan but I'm not completely comfortable with this change (if we do keep it we might as well remove the struct completely). I think having this be something we can expose through CVars or the Settings Registry might be a better idea.

Reply:
I see, these variables are declared and never modified in the two places they are used. so cVars and settings are user configurable?

Reply:
These variables are for our custom docking behavior/colors, so I don't think we'd want to have separate instances of it with different values. You CAN have separate fancy docking instances (e.g. for main Editor vs. separate tools like Script Canvas, Landscape Canvas, etc...), but they should have a consistent docking behavior.

Reply:
If that's the case then can these all be moved to constants outside the struct, they can just be at file scope (maybe inside a namespace if desired) but having fixes constants in a struct like this which is meant for instance data just feels wrong to me, switching `struct` to `namespace` might be the simplest fix 😝

Reply:
> 

I can do that, seems to make more sense.